lib/tasks/db_maintenance.rake

```diff
@@ -251,6 +250,47 @@ namespace :cartodb do
       }, threads, thread_sleep, database_host)
     end
 
+    desc 'Install/upgrade CARTODB SQL functions - only organization owners for Bloomberg merge code release'
+    task :load_functions_bbg, [:database_host, :version, :num_threads, :thread_sleep, :sleep, :statement_timeout] => :environment do |task_name, args|
+      # Send this as string, not as number
+      extension_version = args[:version].blank? ? nil : args[:version]
+      database_host = args[:database_host].blank? ? nil : args[:database_host]
+      threads = args[:num_threads].blank? ? 1 : args[:num_threads].to_i
+      thread_sleep = args[:thread_sleep].blank? ? 0.25 : args[:thread_sleep].to_f
+      sleep = args[:sleep].blank? ? 1 : args[:sleep].to_i
+      statement_timeout = args[:statement_timeout].blank? ? 180000 : args[:statement_timeout]
+
+      puts "Running extension update with following config:"
+      puts "extension_version: #{extension_version.nil? ? 'UNSPECIFIED/LATEST' : extension_version}"
+      puts "database_host: #{database_host.nil? ? 'ALL' : database_host}"
+      puts "threads: #{threads}"
+      puts "thread_sleep: #{thread_sleep}"
+      puts "sleep: #{sleep}"
+      puts "statement_timeout: #{statement_timeout}"
+
+      if database_host.nil?
+        count = ::User.count
+      else
+        count = ::User.where(database_host: database_host).count
+      end
+      execute_on_users_with_index(task_name, Proc.new { |user, i|
+        begin
+          log(sprintf("Trying on %-#{20}s %-#{20}s (%-#{4}s/%-#{4}s)...", user.username, user.database_name, i+1, count), task_name, database_host)
+          # Only load cartodb extensions once per database, i.e. for dedicated users and org owners
+          load_cartodb_extension = user.account_type == '[DEDICATED]' || user.organization_owner?
+          if load_cartodb_extension
+            log(sprintf("Loading cartodb extension and rebuild quota on %-#{20}s %-#{20}s (%-#{4}s/%-#{4}s)...", user.username, user.database_name, i+1, count), task_name, database_host)
+          end
+          user.db_service.load_cartodb_functions(statement_timeout, extension_version, load_cartodb_extension, load_cartodb_extension)
```

Comment:
I don't think passing false for `rebuild_quota` from here is the right way to optimize this function.  This code will still spin off a thread and load user objects from the db for every single user. I think adding an additional filter to `execute_on_users_with_index` to only execute this block for users who are db owners is the way to go here.

Reply:
That is a reasonable approach.  If this is something to be used upstream and more than the initial merge upgrade then it makes sense to modify this further.  If we put the user check outside of the loop then both rake tasks should be modified with this logic and the load_cartodb_extension and rebuild_quota parameters of load_cartodb_functions() should be removed.